mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td

```diff
@@ -222,9 +222,60 @@ def LinalgFillOpInterface : OpInterface<"FillOpInterface"> {
   ];
 }
 
+def DomainAndOperandsAffineMapTransferInterface 
+    : OpInterface<"DomainAndOperandsAffineMapTransferInterface"> {
+  let description = [{
+    Interface for operations that connect an iteration domain to operands via
+    affine maps. Provides methods to access indexing maps between iteration
+    domain and operand index spaces.
+  }];
+  let cppNamespace = "::mlir::linalg";
```

Comment:
Does this need to live in linalg dialect? Can we move this out to a seperate interface like TilingInterface? This is more general I think.

Can be done in a seperate patch.

Reply:
Yeah it's the plan to split things out, we still need a place for "structured stuff" that is not Linalg.